sonaranalyzer-dotnet/build/build-utils.ps1

```diff
@@ -0,0 +1,330 @@
+}
+
+function Get-BranchName {
+    if ($env:GITHUB_BRANCH) {
+        if ($env:GITHUB_BRANCH.StartsWith("refs/heads/")) {
+            return $env:GITHUB_BRANCH.Substring(11)
+        }
+        return $env:GITHUB_BRANCH
+    }
+    return Exec { & git rev-parse --abbrev-ref HEAD }
+}
+
+function Get-Sha1 {
+    if ($env:GIT_SHA1) {
+        return $env:GIT_SHA1
+    }
+    return Exec { & git rev-parse HEAD }
+}
+
+function Get-ExecutablePath([string]$name, [string]$envVar) {
+    $path = [environment]::GetEnvironmentVariable($envVar, "Process")
+
+    if (!$path) {
+        $path = Exec { & where.exe $name } `
+            | Select-Object -First 1
+    }
+
+    if (Test-Path $path) {
+        return $path
+    }
+
+    Write-Error "Cannot find $name"
+    exit 1
+}
+
+function Get-NuGetPath {
+    $path = Get-ExecutablePath -name "nuget.exe" -envVar "NUGET_PATH"
+    Write-Host "Found nuget.exe at ${path}"
+    return $path
+}
+
+function Get-MsBuildPath {
+    $path = Get-ExecutablePath -name "msbuild.exe" -envVar "MSBUILD_PATH"
+    Write-Host "Found msbuild.exe at ${path}"
+    return $path
+}
+
+function Get-VsTestPath {
+    $path = Get-ExecutablePath -name "VSTest.Console.exe" -envVar "VSTEST_PATH"
+    Write-Host "Found VSTest.Console.exe at ${path}"
+    return $path
+}
+
+function Expand-ZIPFile($source, $destination) {
+    Write-Host "Expanding ZIP file ${source}"
+    $application = New-Object -Com Shell.Application
+    $zip = $application.NameSpace($source)
+    foreach ($item in $zip.items()) {
+        $application.NameSpace($destination).CopyHere($item, 0x14)
+    }
+}
+
+function Get-SonarQubeRunnerPath {
+    $sonarqube_runner_exe = (Resolve-RepoPath "MSBuild.SonarQube.Runner.exe")
+
+    if (Test-Path $sonarqube_runner_exe) {
+        return $sonarqube_runner_exe
+    }
+
+    $downloadLink = "https://github.com/SonarSource-VisualStudio/sonar-msbuild-runner/releases/download/${sonarqube_runner_version}/MSBuild.SonarQube.Runner-${sonarqube_runner_version}.zip"
+
+    $sonarqube_runner_zip = (Resolve-RepoPath "MSBuild.SonarQube.Runner.zip")
+
+    (New-Object System.Net.WebClient).DownloadFile($downloadLink, $sonarqube_runner_zip)
+
+    # perhaps we could use other folder, not the repository root
+    Expand-ZIPFile $sonarqube_runner_zip (Resolve-RepoPath "")
+    # PS v5.0 -> Expand-Archive $sonarqube_runner_zip (Resolve-RepoPath "") -Force
+
+    Remove-Item $sonarqube_runner_zip -Force
+
+    Write-Host "Found MSBuild.SonarQube.Runner.exe at ${sonarqube_runner_exe}"
+
+    return $sonarqube_runner_exe
+}
+
+function Set-Version {
+    Write-Header "Setting version..."
+
+    $buildNumber = Get-BuildNumber
+    $branchName = Get-BranchName
+    $sha1 = Get-Sha1
+
+    Write-Host "Setting build number ${buildNumber}, sha1 ${sha1} and branch ${branchName}"
+
+    $versionPropsPath = (Resolve-RepoPath "build\Version.props")
+
+    (Get-Content $versionPropsPath) `
+ -Replace '<Sha1>.*</Sha1>', "<Sha1>$sha1</Sha1>" `

```

Comment:
It is not, but VS Code does it wrong when I reformat document and I am tired of fixing it...